core/market/src/negotiation/common.rs

```diff
@@ -479,64 +531,130 @@ impl CommonBroker {
                 }
                 _ => {
                     // TODO: Don't leak our database error, but send meaningful message as response.
-                    log::warn!("[ProposalReceived] Error saving Proposal: {}", e);
-                    RemoteProposalError::Unexpected(e.to_string())
+                    let msg = format!("Failed saving Proposal [{}]: {}", proposal.body.id, e);
+                    log::warn!("{}", msg);
+                    ProposalValidationError::Internal(msg).into()
                 }
             })?;
 
         // Create Proposal Event and add it to queue (database).
         // TODO: If creating Proposal succeeds, but event can't be added, provider
         // TODO: will never answer to this Proposal. Solve problem when Event API will be available.
         let subscription_id = proposal.negotiation.subscription_id.clone();
-        let proposal = self
-            .db
+        self.db
             .as_dao::<NegotiationEventsDao>()
-            .add_proposal_event(proposal, owner)
+            .add_proposal_event(&proposal, caller_role.swap())
             .await
             .map_err(|e| {
                 // TODO: Don't leak our database error, but send meaningful message as response.
-                log::warn!("[ProposalReceived] Error adding Proposal event: {}", e);
-                RemoteProposalError::Unexpected(e.to_string())
+                let msg = format!("Failed adding Proposal [{}] Event: {}", proposal.body.id, e);
+                log::warn!("{}", msg);
+                ProposalValidationError::Internal(msg)
             })?;
 
         // Send channel message to wake all query_events waiting for proposals.
         self.negotiation_notifier.notify(&subscription_id).await;
 
-        match owner {
-            OwnerType::Requestor => counter!("market.proposals.requestor.received", 1),
-            OwnerType::Provider => counter!("market.proposals.provider.received", 1),
+        match caller_role {
+            Owner::Requestor => counter!("market.proposals.requestor.received", 1),
+            Owner::Provider => counter!("market.proposals.provider.received", 1),
```

Comment:
Important: You swapped semantic `owner` -> `caller_role`, but you didn't swapped counters?

Reply:
Yes, and it is on purpose. Consider it for a bit more time ;)
We want to report, that (local or remote) provider has rejected proposal, and it was done on this node.

Reply:
fixed